CDCgov/prime-reportstream
# 10510: remove support for relative pathing and schema setting normalization

This PR adds the SOP for managing translations that exist in azure and performs final code cleanups for the feature.

Test Steps:
1. Run `./gradlew fatjar`
2. Spin up report stream
3. Make sure that you have both azurite instances running
4. In the STAGING instance in the metadata container, create the directory structure `hl7_mapping/receivers/STLTs/CO/` and then upload the input.fhir, output.hl7 and sender-transform.yml file in `prime-router/src/test/resources/fhirengine/translation/FHIR_to_HL7`, but rename sender-transform.yml to CO.yml
5. In `hl7_mapping` create a `valid-{TIMESTAMP}.txt` (valid-2023-03-29T23:55:00.488Z.txt) and `previous-valid-{TIMESTAMP minus five minutes}.txt` (previous-valid-2023-03-25T23:50:00.488Z.txt)
6. Verify that the in other azurite container, in metadata, the `hl7_mapping` directory does not exist
7. Run `./prime validateSchemas --schema-type="HL7" --blob-store-connect="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:11000/devstoreaccount1;" --blob-store-container="metadata"` to verify the setup.  Everything should be valid
8. Run `./prime syncSchemas --schema-type="HL7" -sb="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:11000/devstoreaccount1;" -sc="metadata" -db="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1;" -dc="metadata"`
9. Verify that the validate function triggers and the azurite container that was empty now contains the schemas in hl7_mapping and the validating.txt file has been removed
10. Edit the CO.yml and output.hl7 file (I changed the MSH.5.1 mapping from PURPLE to BLUE)
11. Turn off report stream and in [build.gradle](https://github.com/CDCgov/prime-reportstream/blob/8038f4fc862e44faf76d95ddb0a5dad4603ef149/prime-router/build.gradle.kts#L99), change to point to the azurite stage container
12. Turn report stream back on
13. Run `./prime syncSchemas --schema-type="HL7" -db="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:11000/devstoreaccount1;" -sc="metadata" -sb="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1;" -dc="metadata"` and verify that the azurite stage schema gets updated and the validating.txt is removed
14. Apply the following patch and reload settings
```
diff --git a/prime-router/settings/organizations.yml b/prime-router/settings/organizations.yml
index c3e42d7dd..629fb44d1 100644
--- a/prime-router/settings/organizations.yml
+++ b/prime-router/settings/organizations.yml
@@ -226,7 +226,7 @@
         numberPerDay: 1440 # Every minute
         initialTime: 00:00
       translation: !<HL7>
-        schemaName: "classpath:/metadata/hl7_mapping/receivers/STLTs/CO/CO.yml"
+        schemaName: "azure:/hl7_mapping/receivers/STLTs/CO/CO.yml"
         useTestProcessingMode: false
         useBatchHeaders: true
         type: "HL7"
```
15. Verify that the smoke tests all pass

## Changes
- Creates the SOP for managing translation schemas outside of deployment cycles
- Removes old code that used relative pathing to download schemas
- Adds functionality to initialize an empty translation azure container

## Checklist

### Testing

### Process

## Linked Issues
- #10510 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
